File: js/problem9.js
//solution: 31875000
print(eulerProblem9());

function eulerProblem9() {
    for(var i = 1;; ++i) {
        for(var j = i+1; i+j < 1000; ++j) {
            var k = 1000 - j - i;
            if(k*k == j*j + i*i) {
                return i*j*k;
            }
        }
    }
}

File: python/problem9.py
#!/usr/bin/python
#solution: 31875000

import math

def euler_problem_9():
    for a in range(1, 1000, 1):
        for b in range(1, 1000, 1):
            for c in range(1, 1000, 1):
                if a + b + c == 1000 and int(math.pow(a, 2)) + int(math.pow(b, 2)) == int(math.pow(c, 2)):
                    return a * b * c

print euler_problem_9()
